freqtrade/strategy/interface.py

```diff
@@ -276,19 +276,25 @@ def get_signal(self, pair: str, interval: str,
             return False, False
 
         latest = dataframe.iloc[-1]
-
-        # Check if dataframe is out of date
         signal_date = arrow.get(latest['date'])
         interval_minutes = timeframe_to_minutes(interval)
+
+        # Check if dataframe is out of date
         offset = self.config.get('exchange', {}).get('outdated_offset', 5)
         if signal_date < (arrow.utcnow().shift(minutes=-(interval_minutes * 2 + offset))):
             logger.warning(
                 'Outdated history for pair %s. Last tick is %s minutes old',
                 pair,
-                (arrow.utcnow() - signal_date).seconds // 60
+                int((arrow.utcnow() - signal_date).total_seconds() // 60)
             )
             return False, False
 
+        # Check if dataframe has new candle
+        if (arrow.utcnow() - signal_date).total_seconds() // 60 >= interval_minutes:
+            logger.warning('Old candle for pair %s. Last candle is %s minutes old',
+                           pair, int((arrow.utcnow() - signal_date).total_seconds() // 60))
+            return False, False
```

Comment:
this will produce very many logs - especially on lower timeframes (1m / 5m on non-highvolume pairs)  as this will happen at almost every candle (switch).

the endresult of this is basically the same as the comparison above for outdated history ... but without being configurable, and without the 2x offset - and with the sides inversed.

and it's really not fixing the problem we discussed yesterday anyway - as that was to do with locked pairs beeing unlocked too early.

This behaviour can be mimict by setting  `exchange.outdated_offset` to `-interval_minutes` (depends on your timeframe, obviously) - and will have the same exact effect - just with a slightly different error message.

Reply:
The warning message usually occurs only for pairs that still hasn't produced a new candle (as discussed yesterday), but perhaps we should rephrase it to make users panic less about it? Or switch it to a debug all-together.

It does fix the problem from a different angle, because I figured after our discussion yesterday that pairs getting unlocked only prevents duplicate buys on the same candle, but it still doesn't fix it for candles that were not bought/locked before.

The whole purpose of this, is to fix signals being produced out of older candles, and yes, I noticed using outdated_offset would fix it, but I noticed it's not exactly the documented purpose for that feature, this one is just to confirm receiving new candles before producing a signal.

So what do you suggest?

Reply:
nothing at the moment ... i'll need to think about this and time to play with it to see the effects it has for different exchanges, different pairs and different timeframes.

I'll try to get to it soon - but it'll probably be weekend ...